partner_lastname_uppercase/models/res_config_settings.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 Coop IT Easy SC
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    uppercase_lastname_activated = fields.Boolean(
+        default=False,
+        config_parameter="uppercase_lastname_activated",
+    )
```

Comment:
i don’t like the name so much. i would prefer something like `convert_lastnames_to_uppercase`. look at the other boolean settings in odoo.

moreover, most prefix the `config_parameter` with the module name.